internal/translator/translate_test.go

```diff
@@ -76,36 +76,70 @@ func TestTranslateIngressRoute(t *testing.T) {
 
 func TestTranslateIngressRouteErrors(t *testing.T) {
 
-	nonrootIngressRoute := []byte(`
+	tests := map[string]struct {
+		input []byte
+		want  []string
+	}{
+		"nonroot IR, multiple invalid paths": {
+			input: []byte(`
+---
 apiVersion: contour.heptio.com/v1beta1
 kind: IngressRoute
-metadata: 
-  name: nonroot
+metadata:
+  name: nonroot-invalid-badpaths
   namespace: default
-spec: 
-  routes: 
-  - match: /
-    services: 
-    - name: s2
-      port: 80
-`)
-
-	fatalError := []string{"unimplemented: Can't translate non-root IngressRoutes yet"}
-
-	ir, err := k8sdecoder.DecodeIngressRoute(nonrootIngressRoute)
-	if err != nil {
-		t.Fatal(err)
+spec:
+  routes:
+    - match: foo
+      services:
+        - name: s1
+          port: 80
+    - match: bar
+      services:
+        - name: s2
+          port: 80
+`),
+			want: []string{"invalid IngressRoute: match clauses must share a common prefix"},
+		},
+		"nonroot IR, multiple nonmatching prefixes": {
+			input: []byte(`
+---
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata:
+  name: nonroot-invalid-badpaths
+  namespace: default
+spec:
+  routes:
+    - match: /foo
+      services:
+        - name: s1
+          port: 80
+    - match: /bar
+      services:
+        - name: s2
+          port: 80
+`),
+			want: []string{"invalid IngressRoute: match clauses must share a common prefix"},
+		},
 	}
-	_, _, err = IngressRouteToHTTPProxy(ir)
-	if err != nil {
-		// Can't translate the IngressRoute at all
-		// errors.txt should have the error message.
-		errorDiff := cmp.Diff([]string{err.Error()}, fatalError)
-		if errorDiff != "" {
-			t.Fatalf("Unexpected translation failure:\n%v", errorDiff)
-		}
-		// Errors match, pass.
-		return
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ir, err := k8sdecoder.DecodeIngressRoute(tc.input)
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, _, err = IngressRouteToHTTPProxy(ir)
+			if err != nil {
+				// Can't translate the IngressRoute at all
+				// errors.txt should have the error message.
```

Comment:
Is it reasonable to not have to commit the empty errors files?

Reply:
Dave and I went back and forth on that. I originally had it that no error file equals no errors expected, but Dave felt that enforcing that all the fixtures look exactly the same was better. I didn't feel it was a hill worth dying on.

Reply:
Yup, fair enough.